modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/RDBMSCoordinationStrategy.java

```diff
@@ -212,6 +212,23 @@ public void start(CoordinationConfigurableClusterAgent configurableClusterAgent,
 
         coordinatorElectionTask = new CoordinatorElectionTask();
         threadExecutor.execute(coordinatorElectionTask);
+
+        // Wait until node state become Candidate/Coordinator because thrift server needs to start after that.
+        int waitTime = 0;
+        int maxWaitTime = heartbeatMaxAge * 5;
+        while (currentNodeState == NodeState.ELECTION) {
+            try {
+                TimeUnit.MILLISECONDS.sleep(500);
+                waitTime = waitTime + 500;
+                if (waitTime == maxWaitTime) {
+                    throw new RuntimeException("Node status didn't change to Candidate/Coordinator within "
+                            + waitTime + " milliseconds.");
```

Comment:
Maybe we can say something similar to "Node is stuck in the election for ___ milliseconds". In the future, we might add some more states other than candidate and coordinator